### ISSUE TITLE
The value Acc of the test result is nan, Dice and Iou are 0

### ISSUE THREAD

**Issue by Kuanzhaipcx** [open-mmlab/mmsegmentation#3449]
The validation result of the model is all right, but in the test time there is a situation that the indicator of the tongue class is nan and 0, I have tried many methods but no solution, if I can get your help I will be grateful!
![image](https://github.com/open-mmlab/mmsegmentation/assets/124220581/ed17bc15-5bb2-4752-a8cb-fdf82516b70a)

Here is my dataset configuration file：
dataset_type = 'Tongue_public' 
data_root = './data/TongueImage' 
 
img_norm_cfg = dict( 
    mean=[113.536, 91.263, 83.868], std=[46.397, 45.856, 42.823], to_rgb=True) 
 
crop_size = (512, 512)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', reduce_zero_label=False),
    dict(
        type='RandomResize',
        scale=(2048, 512),
        ratio_range=(0.5, 2.0),
        keep_ratio=True),
    dict(type='RandomCrop', crop_size=crop_size, cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(type='PackSegInputs')
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='Resize', scale=(2048, 512), keep_ratio=True),
    # add loading annotation after ``Resize`` because ground truth
    # does not need to do resize data transform
    dict(type='LoadAnnotations', reduce_zero_label=False),
    dict(type='PackSegInputs')
]
train_dataloader = dict(
    batch_size=4,
    num_workers=4,
    persistent_workers=True,
    sampler=dict(type='InfiniteSampler', shuffle=True),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_prefix=dict(
            img_path='images/train', seg_map_path='annotations/train'),
        pipeline=train_pipeline))
val_dataloader = dict(
    batch_size=1,
    num_workers=4,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_prefix=dict(
            img_path='images/val',
            seg_map_path='annotations/val'),
        pipeline=test_pipeline))
test_dataloader = dict(
    batch_size=2,
    num_workers=4,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        data_prefix=dict(
            img_path='images/test',
            seg_map_path='annotations/test'),
        pipeline=test_pipeline))

val_evaluator = dict(type='IoUMetric', iou_metrics=['mDice','mIoU'])
test_evaluator = val_evaluator

**Comment by xiexinch** [comment]
Hi @Kuanzhaipcx,
Please follow the [FAQ doc](https://mmsegmentation.readthedocs.io/en/latest/notes/faq.html#how-to-handle-binary-segmentation-task) to configure your experiment. This result is usually caused by the wrong setting of the binary segmentation task.

**Comment by Kuanzhaipcx** [comment]
> Hi @Kuanzhaipcx, Please follow the [FAQ doc](https://mmsegmentation.readthedocs.io/en/latest/notes/faq.html#how-to-handle-binary-segmentation-task) to configure your experiment. This result is usually caused by the wrong setting of the binary segmentation task.

Thanks for your reply, but I would like to know why the metrics are calculated normally on the validation set?
![image](https://github.com/open-mmlab/mmsegmentation/assets/124220581/2090b885-ae26-4175-b9f9-8975a831a6e6)

**Comment by xiexinch** [comment]
I'm sorry that from the info you provided, I can't locate the problem. I think you can evaluate the test set and validation set separately to check if it's a problem with the data or the test script.

**Comment by Kuanzhaipcx** [comment]
> I'm sorry that from the info you provided, I can't locate the problem. I think you can evaluate the test set and validation set separately to check if it's a problem with the data or the test script.

Sorry to bother you again, but I've done two things recently:
1. Change the path of the test set to the verification set, and the result indicators can be displayed normally.
2. Copied some images from the verification set to the test set, and the result indicators can also be displayed normally.
I guarantee that there is no problem with my test data, because they are split from the same data set as the training set and validation set. I wonder if the problem is caused by too little test data?

**Comment by xxxxyliu** [comment]
> > I'm sorry that from the info you provided, I can't locate the problem. I think you can evaluate the test set and validation set separately to check if it's a problem with the data or the test script.
> 
> Sorry to bother you again, but I've done two things recently:
> 
> 1. Change the path of the test set to the verification set, and the result indicators can be displayed normally.
> 2. Copied some images from the verification set to the test set, and the result indicators can also be displayed normally.
>    I guarantee that there is no problem with my test data, because they are split from the same data set as the training set and validation set. I wonder if the problem is caused by too little test data?

I met the same problem！Has the problem been resolved?

**Comment by xxxxyliu** [comment]
It might be due to too few images in the test or validation set, and insufficient training of the network in the early stages.my val just have 1 image.so..

**Comment by Rainbow-cocktail** [comment]
> Hi @Kuanzhaipcx, Please follow the [FAQ doc](https://mmsegmentation.readthedocs.io/en/latest/notes/faq.html#how-to-handle-binary-segmentation-task) to configure your experiment. This result is usually caused by the wrong setting of the binary segmentation task.

Thank you for your reply. For additional information, I strongly recommend following the second method provided in the FAQ document. It is quite useful.
